src/lint/linter/UberCheckstyleLinter.php

```diff
@@ -5,270 +5,338 @@
+    private $useScript = false;
+    private $maxFiles = 100;
+
+    public function getInfoName() {
+        return 'Java checkstyle linter';
     }
-    return parent::setLinterConfigurationValue($key, $value);
-  }
-
-  /**
-   * Check that the checkstyle libraries are installed.
-   * @return void
-   */
-  private function checkConfiguration() {
-    if ($this->checkstyleJar != null) {
-      $this->checkJarConfiguration();
-    } else if ($this->checkstyleScript != null) {
-      $this->checkScriptConfiguration();
-      $this->useScript = true;
-    } else {
-      throw new ArcanistMissingLinterException(
-        pht('Missing config.  Either \'checkstyle.script\' or \'checkstyle.jar\' need to be set. ')
-      );
+
+    public function getLinterName() {
+        return 'CHECKSTYLE';
+    }
+
+    public function getInfoURI() {
+        return 'http://checkstyle.sourceforge.net';
     }
-  }
-
-  private function checkScriptConfiguration() {
-    // Some scripts have command line arguments
-    $script = explode(' ', $this->checkstyleScript)[0];
-    $absScriptPath = Filesystem::resolvePath($script, $this->getProjectRoot());
-    if (!Filesystem::pathExists($absScriptPath)) {
-      throw new ArcanistMissingLinterException(
-        pht(
-          'Unable to locate script "%s" to run linter %s. You may need ' .
-          'to install the script, or adjust your linter configuration.',
-          $absScriptPath,
-          get_class($this)));
+
+    public function getInfoDescription() {
+        return 'Use checkstyle to perform static analysis on Java code';
     }
-  }
 
-  private function checkJarConfiguration() {
-    if (!Filesystem::binaryExists("java")) {
-      throw new ArcanistMissingLinterException(
-        pht('Java is not installed', get_class($this)));
+    public function getLinterConfigurationName() {
+        return 'checkstyle';
     }
 
-    $absJarPath = Filesystem::resolvePath($this->checkstyleJar, $this->getProjectRoot());
-    if (!Filesystem::pathExists($absJarPath)) {
-      throw new ArcanistMissingLinterException(
-        sprintf(
-          "%s\n%s",
-          pht(
-            'Unable to locate jar "%s" to run linter %s. You may need ' .
-            'to install the script, or adjust your linter configuration.',
-            $absJarPath,
-            get_class($this)),
-          pht('
-                TO INSTALL: 
-                1) Download checkstyle jar from https://sourceforge.net/projects/checkstyle/files/checkstyle/
-                2) Set `checkstyle.jar` in `.arclint` to the location of the jar.'
-          )));
+    public function getLinterConfigurationOptions() {
+        $options = array(
+            'checkstyle.script' => array(
+                'type' => 'optional string',
+                'help' => pht('Checkstyle script to execute. Script must output checkstyle in XML to $stdout'),
+            ),
+            'checkstyle.jar' => array(
+                'type' => 'string',
```

Comment:
This should be optional as some repos have the checkstyle script autodownload the jar